flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableSource.java

```diff
@@ -160,7 +162,14 @@ private void initAllPartitions() {
 						String partitionValue = partition.getValues().get(i);
 						partitionSpec.put(partitionColName, partitionValue);
 						DataType type = catalogTable.getSchema().getFieldDataType(partitionColName).get();
-						Object partitionObject = restorePartitionValueFromFromType(partitionValue, type);
+						Object partitionObject;
+						if (defaultPartitionName.equals(partitionValue)) {
+							LogicalTypeRoot typeRoot = type.getLogicalType().getTypeRoot();
+							// while this is inline with Hive, seems it should be null for string columns as well?
```

Comment:
Hive also has String type. What's Hive's behavior on this?

Reply:
For a string partition column, it'll be assigned the default value, e.g. `__HIVE_DEFAULT_PARTITION__`. But I thought it should simply be null, just like other kinds of partition columns.